### PR TITLE
Implement scoped CSS with shadow-dom

### PR DESCRIPTION
**Pull request by ragingwind**
Implement  stylesheet in QTI spec. To make done this, Shadow DOM was featured with React.

- The stylesheet must be isolated and scoped
- Other CSS features should work well

### CODE REVIEW

File: packages/react/src/QtiViewer.tsx [hunk 1]
```diff
@@ -1,24 +1,40 @@
...
+}
 
-  if (!itemBody) {
-    throw new Error('QTI itemBody is not found');
-  }
+type ReactShadowRoot = ShadowRoot | Element;
 
-  return <>{Qti.parseXml(itemBody)}</>;
-});
+const ShadowRoot: React.FC<ShadowRootProp> = ({assessmentItem}) => {
```
Comment (Il9) [inline packages/react/src/QtiViewer.tsx:20]: 굳이 name을 ShadowRoot로 지어야 될까 싶네요. 그냥 root이고 shadowDOM을 내부에서 사용만 하면 되지 않을까요?
뭔가 styleSheet만을 위한 컴포넌트로 보이지 않을까 싶네요

File: packages/react/src/QtiViewer.tsx [hunk 2]
```diff
@@ -1,24 +1,40 @@
...
+  useEffect(() => {
+    const shadow = parentRef.current?.attachShadow({
+      mode: 'open',
+    });
+
+    if (shadow) {
+      assessmentItem.stylesheets.forEach(s => shadow.appendChild(s));
+      setShadowRoot(shadow);
+    }
+  }, [parentRef, assessmentItem.stylesheets]);
```
Comment (Il9) [inline packages/react/src/QtiViewer.tsx:34]: 이건 비동기가 아니라면 그냥 바로 기본값으로 넣어주면 되지 않을까요?

File: packages/react/src/QtiViewer.tsx (hunk 1 above)
Comment (ragingwind) [reply packages/react/src/QtiViewer.tsx:20]: Actually, children will be attached to `Shadow Root` in the component, https://developer.mozilla.org/en-US/docs/Web/Web_Components/Using_shadow_DOM

File: packages/react/src/QtiViewer.tsx (hunk 2 above)
Comment (ragingwind) [reply packages/react/src/QtiViewer.tsx:34]: Which value you addressed? shadow?

File: packages/react/src/QtiViewer.tsx (hunk 1 above)
Comment (ragingwind) [reply packages/react/src/QtiViewer.tsx:20]: It's more clear to me

Comment (Il9) [reply packages/react/src/QtiViewer.tsx:20]: Okay

File: packages/react/src/QtiViewer.tsx (hunk 2 above)
Comment (Il9) [reply packages/react/src/QtiViewer.tsx:34]: This means shadow.
But it's vague because it seems like there might be no shadow

Comment (Il9) [reply packages/react/src/QtiViewer.tsx:34]: I think you can just do it like this

File: packages/react/src/interactions/ChoiceInteraction/ChoiceInteraction.stories.tsx [hunk 3]
```diff
@@ -9,3 +9,12 @@ export const choice = QtiViewerTemplate.bind({});
...
   assessmentItemSrc: 'items/choice.xml',
 };
+
+export const orkney1 = QtiViewerTemplate.bind({});
+
+orkney1.args = {
+  assessmentItemSrc: 'items/orkney1.xml',
+  onChange: (value: any) => {
+    console.log(value);
+  },
```
[the remark this answers is not in the document]
Comment (ragingwind) [reply packages/react/src/interactions/ChoiceInteraction/ChoiceInteraction.stories.tsx:19]: solved from https://github.com/riiid/qtikit/commit/5891ef86e2b1f565b5c827d43ff4c476b4a9abd7

**Pull request merged**